File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/ClientDetails.java
package kz.greetgo.sandbox.controller.model;


import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class ClientDetails {
  public String id;
  public String name;
  public String surname;
  public String patronymic;
  public String gender;
  public String charm;
  public String  birthDate;
  public AddressInfo regAddress = new AddressInfo();
  public AddressInfo factAddress = new AddressInfo();
  public List<PhoneInfo> phones = new ArrayList<>();

  @Override
  public String toString() {
    return "ClientDetails{" +
      "id='" + id + '\'' +
      ", name='" + name + '\'' +
      ", surname='" + surname + '\'' +
      ", patronymic='" + patronymic + '\'' +
      ", gender='" + gender + '\'' +
      ", charm='" + charm + '\'' +
      ", birthDate='" + birthDate + '\'' +
      ", regAddress=" + regAddress +
      ", factAddress=" + factAddress +
      ", phones=" + phones +
      '}';
  }
}

File: sandbox.client/front/ts/model/ClientDetails.ts
import {AddressInfo} from "./AddressInfo";
import {PhoneInfo} from "./PhoneInfo";
export class ClientDetails {
    public id : string;
    public name : string;
    public surname : string;
    public patronymic : string;
    public gender : string;
    public charm : string;
    public birthDate : Date;
    public regAddress : AddressInfo;
    public factAddress : AddressInfo;
    public phones : Array<PhoneInfo>;


    public assign(o: ClientDetails): ClientDetails {
        this.id = o.id;
        this.name = o.name;
        this.surname = o.surname;
        this.patronymic = o.patronymic;
        this.gender = o.gender;
        this.charm = o.charm;
        this.birthDate = o.birthDate;
        this.regAddress = o.regAddress;
        this.factAddress = o.factAddress;
        this.phones = o.phones;

        return this;
    }
}
